src/generator/type_inference_visitor.ts

```diff
@@ -66,7 +72,7 @@ class TypeInferenceVisitor extends Visitor {
     protected visitNumericLiteral(node: NumericLiteral) {
         super.visitNumericLiteral(node);
 
-        this.expressionTypes.set(node, WebAssemblyType.INT_32);
+        this.expressionTypes.set(node, getWebAssemblyType(node.value));
```

Comment:
The function `getWebassemblyType` also checks for arrays and booleans which is not necessary here.